### PR TITLE
Add tests for guild tooltip generation

### PR DESCRIPTION
**Pull request by h3ndrk**
This PR adds 32 tests for testing the guild tooltip generation. The test requirements have been generated with the ACTS tool.

The tests with the ugly names are the tests for the task's submission. After that, I wrote another test that should work much better and should be better maintainable. I tried writing many comments explaining the automatic test value generation for the improved ACTS test.

FYI: All my tests using ACTS as a tool for generation test requirements (not test cases). Therefore the test value generation happens dynamically in the test case. :sunglasses: 

Run the new tests with (assuming a running MySQL instance, see `README.md`):

```
cargo test --features integration guild_tooltip
```

### CODE REVIEW

File: Backend/src/modules/tooltip/tests/guild_tooltip.rs [hunk 1]
```diff
@@ -0,0 +1,910 @@
...
+        material::{Character, CharacterHistory, Guild},
+        tools::GetGuild,
+        Armory,
+    },
+    tooltip::{tools::RetrieveGuildTooltip, Tooltip},
+};
+
+fn build_character_guild(guild_id: u32) -> CharacterGuild {
+    CharacterGuild {
+        guild_id: guild_id,
```
Comment (Geigerkind) [inline Backend/src/modules/tooltip/tests/guild_tooltip.rs:13]: I suggest you to get clippy as integration. It helps you to not do non idiomatic patterns. This for example can be simplified by simply `guild_id,`

**Comment by h3ndrk** [comment]
> Why do you need it to run under "integration"? You don't use the DB do you?

The reason for the tests being integration tests might be that a full database connection is constructed and this is not optional.

We discussed that already in our chat at 2020-05-15 at ~16:00. Since I inspired myself from the `character_tooltip` module (which is also an integration test), I did not question the fact that all tooltip tests are integration tests. IMHO, changing this is out of the scope of this PR.

File: Backend/src/modules/tooltip/tests/guild_tooltip.rs [hunk 2]
```diff
@@ -0,0 +1,910 @@
...
+    let tooltip = Tooltip::default().init();
+    let tooltip_res = tooltip.get_guild(&armory, 2);
+    assert!(!tooltip_res.is_ok());
+}
+
+#[test]
+fn a2_b1_c1_d1_e1_f1_g3_h2_i3_j2_k3_l2_m2_n3_o1_p1() {
+    let armory = Armory::default();
+    armory.characters.write().unwrap().insert(0, build_character(false, false, 0));
+    armory.guilds.write().unwrap().insert(0, build_guild(0, "".to_string()));
```
[the remark this answers is not in the document]
Comment (h3ndrk) [reply Backend/src/modules/tooltip/tests/guild_tooltip.rs:505]: Yes, but if possible, I don't want to touch these ugly tests anymore. :wink:

File: Backend/src/modules/tooltip/tests/guild_tooltip.rs [hunk 3]
```diff
@@ -0,0 +1,910 @@
...
+
+#[test]
+fn a2_b2_c2_d2_e1_f1_g1_h1_i1_j1_k1_l1_m1_n1_o2_p2() {
+    let armory = Armory::default();
+    armory.characters.write().unwrap().insert(0, build_character(true, true, 0));
+    armory.guilds.write().unwrap().insert(0, build_guild(0, "".to_string()));
+    armory.guilds.write().unwrap().insert(1, build_guild(1, "".to_string()));
+    armory.guilds.write().unwrap().insert(2, build_guild(2, "".to_string()));
+    armory.guilds.write().unwrap().insert(4294967295, build_guild(4294967295, "".to_string()));
+    let tooltip = Tooltip::default().init();
```
[the remark this answers is not in the document]
Comment (h3ndrk) [reply Backend/src/modules/tooltip/tests/guild_tooltip.rs:526]: This is inspired by your own test:

https://github.com/Geigerkind/LegacyPlayersV3/blob/34254cd42de37bea2c64fe4004ea73d5741d3644/Backend/src/modules/tooltip/tests/character_tooltip.rs#L13

But I fixed my test. You may want to fix your own. :wink:

File: Backend/src/modules/tooltip/tests/guild_tooltip.rs (hunk 1 above)
Comment (h3ndrk) [reply Backend/src/modules/tooltip/tests/guild_tooltip.rs]: Fixed, I cannot see any other warning in my module that clippy throws.

File: Backend/src/modules/tooltip/tests/guild_tooltip.rs [hunk 4]
```diff
@@ -0,0 +1,910 @@
...
+                        test_requirement[10],
+                        test_requirement[11],
+                        test_requirement[12],
+                        test_requirement[13],
+                        test_requirement[14],
+                        test_requirement[15]
+                    );
+                },
+                _ => {
+                    // test_requirement[15] = 2
```
[the remark this answers is not in the document]
Comment (h3ndrk) [reply Backend/src/modules/tooltip/tests/guild_tooltip.rs]: Added, fixed.

**Pull request merged**